packages/remark-math/inline.js

```diff
@@ -6,35 +6,46 @@ const ESCAPED_INLINE_MATH = /^\\\$/ // starts with \$
 
 const NODE_TYPES = {
   TEXT: 'text',
-  MATH_INLINE: 'inlineMath'
+  INLINE_MATH: 'inlineMath'
+}
+
+const includeNonDollarsAndEscaped = ['\\$', /[^$]/]
+
+function escapeRegExp (str) {
```

Comment:
Cool! I like this. Let's leave a note here about where it came from so that it doesn't look like voodoo:

```suggestion
// Taken from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Escaping
function escapeRegExp (str) {
```